packages/text-bitmap/src/BitmapText.ts

```diff
@@ -301,53 +331,199 @@ export class BitmapText extends Container
         }
 
         const lenChars = chars.length;
-        const tint = this.tint;
+
+        const pagesMeshData: Record<number, PageMeshData> = {};
+
+        // new Map<BaseTexture, PageMeshData>();
+        const meshesToAdd: Mesh[] = [];
 
         for (let i = 0; i < lenChars; i++)
         {
-            let c = this._glyphs[i]; // get the next glyph sprite
+            const texture = chars[i].texture;
+            const baseTextureUid = texture.baseTexture.uid;
 
-            if (c)
+            if (!pagesMeshData[baseTextureUid])
             {
-                c.texture = chars[i].texture;
+                let pageMeshData = pageMeshDataPool.pop();
+
+                if (!pageMeshData)
+                {
+                    const geometry = new MeshGeometry();
+                    const material = new MeshMaterial(Texture.EMPTY);
+
+                    const mesh = new Mesh(geometry, material);
+
+                    pageMeshData = {
```

Comment:
heya! The main reason for declaring the whole object is so that we don't change its structure behind the scenes immediately in the next few lines. I think this keeps it a tad more js compiler efficient :D

Reply:
You mean its faster in the browser?